src/main/java/seedu/address/commons/collections/UniqueItemCollection.java

```diff
@@ -31,6 +32,27 @@ protected DuplicateItemException() {
      * Constructs empty ItemCollection.
      */
     public UniqueItemCollection() {}
+    
+    /**
+     * Duplicates an existing UniqueItemCollection
+     */
+    @SuppressWarnings("unchecked")
+	public UniqueItemCollection<T> copyCollection()  {
+    	UniqueItemCollection<T> copiedCollection = new UniqueItemCollection<T>();
+    	for (T item : internalList) {
+    		try {
+	    		if (item instanceof Copiable) {
+	    			Copiable<T> copiableItem = (Copiable<T>) item;
+	    			copiedCollection.add(copiableItem.copy());
+	    		} else {
+	    			assert false : "The items in the list must implement the Copiable interface";
```

Comment:
Would prefer an UnsupportedOperationException here instead of asserts, usually that's used for methods that may not be called under certain implementations